bridges/AppleAppStoreBridge.php

```diff
@@ -78,94 +94,157 @@ public function getName()
         ];
-
-        $json = json_decode(getContents($uri, $headers), true);
-
+        
+        $content = getContents($url, $headers);
+        if (!$content) {
+            throw new \Exception("Failed to get content from API");
+        }
+        
+        $json = json_decode($content, true);
+        
+        if (!isset($json['data']) || empty($json['data'])) {
+            throw new \Exception("No app data found in API response");
+        }
+        
+        $this->debugLog("Successfully retrieved app data from API");
         return $json['data'][0];
     }
 
-    /**
-     * Parses the version history from the data received
-     * @return array list of versions with details on each element
-     */
-    private function getVersionHistory($data, $platform)
+    private function extractAppDetails($data)
     {
-        switch ($platform) {
-            case 'mac':
-                return $data['relationships']['platforms']['data'][0]['attributes']['versionHistory'];
-            default:
-                $os = self::PLATFORM_MAPPING[$platform];
-                return $data['attributes']['platformAttributes'][$os]['versionHistory'];
+        try {
+            if (isset($data['attributes'])) {
+                $this->name = $data['attributes']['name'] ?? null;
+                $author = $data['attributes']['artistName'] ?? null;
+                $this->debugLog("Found app details in attributes: {$this->name} by {$author}");
+                return [$this->name, $author];
+            }
+            
+            $this->name = "App " . $this->getInput('id');
+            $this->debugLog("App details not found, using default: {$this->name}");
+            return [$this->name, "Unknown Developer"];
+        } catch (\Exception $e) {
+            $this->debugLog("Error extracting app details: " . $e->getMessage());
+            $this->name = "App " . $this->getInput('id');
+            return [$this->name, "Unknown Developer"];
         }
     }
 
-    public function collectData()
+    private function getVersionHistory($data)
     {
-        $id = $this->getInput('id');
-        $country = $this->getInput('country');
         $platform = $this->getInput('p');
-
-        switch ($platform) {
-            case 'mac':
-                $data = $this->getDataFromShoebox($id, $platform, $country);
-                break;
-
-            default:
-                $token = $this->getJWTToken($id, $platform, $country);
-                $data = $this->getAppData($id, $platform, $country, $token);
+        $this->debugLog("Extracting version history for platform: {$platform}");
+        
+        try {
+            $platform_key = self::PLATFORM_MAPPING[$platform] ?? $platform;
+            
+            if (isset($data['attributes']['platformAttributes'][$platform_key]['versionHistory'])) {
+                return $data['attributes']['platformAttributes'][$platform_key]['versionHistory'];
+            }
```

Comment:
I think this should work:

`$version = $data['attributes']['platformAttributes'][$platform_key]['versionHistory'] ?? null`